instrumentation/grpc/src/main/java/brave/grpc/GrpcClientResponse.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.grpc;
+
+import brave.Response;
+import brave.Span;
+import brave.internal.Nullable;
+import io.grpc.ClientCall;
+import io.grpc.Metadata;
+import io.grpc.Status;
+
+// intentionally not yet public until we add tag parsing functionality
+final class GrpcClientResponse extends Response {
+  final GrpcClientRequest request;
+  @Nullable final Status status;
+  @Nullable final Metadata trailers;
+  @Nullable final Throwable error;
+
+  GrpcClientResponse(GrpcClientRequest request,
+    @Nullable Status status, @Nullable Metadata trailers, @Nullable Throwable error) {
+    if (request == null) throw new NullPointerException("request == null");
+    this.request = request;
+    this.status = status;
+    this.trailers = trailers;
+    this.error = error != null ? error : status != null ? status.getCause() : null;
```

Comment:
Related to my other comments, instead of passing in an error, for gRPC it's probably more correct to just always use `status.getCause()`. The framework is supposed to convert any business logic exception into a `Status` with the cause set to that exception.

Reply:
I think what you are saying in general is that we should assume grpc will always call close() even if client interceptors/handlers etc fail.

I'm cool with that. We do have code that is in other libraries that we should still be careful about. In the case that we find grpc has bugs what will happen is that we will leak spans. This is imho ok as such leaks we can address in a non-api affecting patch release.

Reply:
I think this comment drifted as we are catching things which don't incur a grpc status, so resolving it.